
Embryonic development has amazed scientists and philosophers for centuries.

Many reasons have been evoked for the perceivable complexity increase that transforms a single cell into a larva or an adult (even when complexity has eluded an unique definition).
Aristotle claimed a "vital force" guided embryogenesis. %Eventually, 
Vitalism was substituted by preformationism and epigenesis explanations in the 18th century.
%
Since the 1980's, when it became clear that genes play a key role in embryogenesis, 
%Nowadays, 
the causal mechanisms for such complexity increase are being searched for at the cellular and molecular level.

Using the number of cell types as complexity measure, its increase during development is self-evident: the embryo begins with one cell type (zygote) and concludes with up to 200 cell types (in mammals). 
This process can also be called embryo compartmentalization.
%From many discoveries since the 1980's, it has become clear that some genes play a key role in the embryo compartmentalization. 
At the level of gene expression, it its assumed that in early development, genes are initially expressed in large domains on the embryo. Then, at later stages, their expression become spatially restricted, until some genes are expressed in a cell-specific manner.
%
For many crucial developmental genes (e.g., Hox genes), the spatio-temporal expression dynamics, and how it relates to the embryo compartmentalization, has been thoroughly described.
It is not clear however, if the dynamics are similar for the rest of the genes deployed in development, or if there are differences between different types of genes.

-----------------------------

Adaptive reasons have been also said to be the cause for the increase in complexity.
A change during development would be adaptive if it increases the fitness of its bearer, even when the effect of this developmental change is realized until the adult or larva.
%The increase in complexity (or any change in development) has been considered for some authors as the adaptations of the embryo to its environment.

Many methods of molecular evolution estimate the action of natural selection based on the quasi-neutral evolution model.
Under this model, an adaptive change that has been caused by a genetic mutation, can be traced in the DNA sequence, as a positive selected site would show less variance than other sites evolving neutrally.

Recently, methods of population genomics use divergence ans polymorphism (inter-specific and intra-specific variation) to measure more precisely the proportion of adaptive mutation at the molecular level.

----------------------------
Importantly, if different developmental stages show distinct levels of positive selection or stabilizing selection,
some specific patterns, when comparing the development trajectories of different species in a group, might be recognizable.
% specific patterns might appear (patterns of what?) when comparing the divergence between different species.
Two related models currently under much discussion? are:
%This is related to already proposed models of divergence: 
1)von Baer's laws and the hourglass model.
The former states that the development of 2 species would be very similar in early stages and increasingly divergent in subsequent stages. In contrast, the latter states that development is less divergent at mid development.

%Such patterns could be due to selection or constrant..
%The print in natural selection on the genome is useful to discern between these scenarios.
%Some regarded each embryonic stage to have adaptation as the result of the constant interaction of the embryo with its environment (whether external or internal).
%Others have distinguished the adaptations in the embryo from those from the adult, stating that the former are product of internal selection, instead of being product of natural selection, as the latter.
------------------------------------------
In here, I will use gene expression information to estimate 
both complexity and adaptation in the embryo.

I will use a statistical approach, this means that no special emphasis on specific genes or pathways would be made. 
Instead, my intention is to provide a broad picture on the spatiotemporal change in complexity and adaptation in the embryo.

%I take advantage of available databases of gene expression.
%I analyze complexity using two popular developmental biology models: Drosophila melanogaster and Ciona intestinalis.
For the estimation of complexity, I analyzed gene expression data (from thousands of in situ hybridization experiments) of two popular developmental biology models: Drosophila melanogaster and Ciona intestinalis (from available databases) and developed quantitative measures of complexity.

To analyze adaptation, I combined the D. melanogaster transcriptomic data with genomic data from the DGRP project. With the DFE-alpha method (which uses coding-region polymorphism data and coding-region divergence between D. yakuba and D. melanogaster to estimate the proportion of adaptive changes), I chart a spatial map of adaptation of the fruit fly embryo's anatomy. 

Also, I analyze the pattern of positive selection through the entire life cycle of D. melanogaster.

Briefly, 

we found that low rates of adaptive change are found in the digestive system and high rates in the gonads and parts of the forming head. We also found that the regions that exhibit the highest rates of adaptation express, on average, genes that are phylogenetically young