The fruit fly, Drosophila melanogaster, has been a great valuable tool for biological research.
Its use as a model system dates back to the beginning of the 20th century.
In 1908, Thomas H. Morgan (see section X) started to grow flies in large quantities to study gene mutations. At that time, the gene concept was an abstract one, as the nature and location of the genes was still disputed.
%
The main advantages of using flies were their rapid generation time, it was easy to culture and cheap to maintain (REF Martinez Arias in Dachmann).
%
In his lab, at the University of Columbia, Morgan encountered a fly with white eyes (the wild-type eye color is red), which became a subject of his research for many years.
Eventually, he discovered that the allele of the gene, that he called \textit{white}, was located in a sex chromosome, demonstrating for the first time the sex-linkage of genes (REF MORGAN 1919).
%
Morgan's students also demonstrated that mutations were inducible with X-rays and introduced the use of "balancer" chromosomes to keep stable stocks of mutants (REF Martinez Arias in Dachmann).
%
The research carried in Morgan's lab laid the basis modern genetics, and its fly room became a central node in the genetics research, establishing Drosophila as a organism model.

However, Drosophila's development was difficult to study, as the embryos were not large enough to experimentally manipulate them, and not transparent enough to visualize with a microscope (REF Gilbert book).


In 1976, E. Lewis published a seminal work, in which he determined the effects of mutations in the Bithorax complex (BX-C).
He determined that the BX-C consisted of distinct genetic elements and there was a correlation in the order of the mutations within the complex and the A/P order of the body affected by them (REF Lewis), a phenomenon called spatial co-linearity.

Lewis discoveries were complemented with the discovery of the Hox genes (REF), a family of transcription factors that was shown to be conserved with vertebrates (REF).
Hox genes in D. melanogaster are arranged in two clusters, the Antennapedia (ANT-C) and the Bithorax cluster.

Molecular biology techniques allowed finally to study fly genes and their effect on embryogenesis, unravelling the mysteries of Drosophila's development 

%This feature, called spatial collinearity, would turn out to be a defining feature of both vertebrate and invertebrate homeotic genes. Lewis showed remarkable vision by arguing that the identity of an individual body segment is produced by the particular combination of BX-C genes, and that these were activated in reponse to an Aâ€“P gradient.

\subsection{D. melanogaster life cycle}

Drosophila melanogaster is a holometabolous insect, which means that it goes through a complete metamorphosis, i.e., the larva and the adult forms are very different.
Its embryonic development is very fast, the larva hatches after around 20 hours (at  X degrees).
The larva grows and passes through two moults before becoming a resting stage called a pupa in which
the body is remoulded to form the adult. 

Much of the adult body is formed
from the imaginal discs and the abdominal histoblasts which are only present
as undifferentiated buds in the larva.

\subsubsection{Developmental stages}

In here, I will briefly summarized the embryonic development of D. melanogaster, for a comprehensive lecture, see (REF Campos-Ortega Hartenstein and Gilbert book)

Campos Ortega has divided the embryonic development of Drosophila in 17 stages. The main events of each stage and its timing under laboratory conditions are presented in Table X.

TABLE


\subsection{Dorso-ventral patterning}

\subsection{Anterio-Posterior patterning}

A milestone on the embryogenesis research on Drosophila took place in 1980, when Eric Wieschaus and Christiane N\"{u}sslein-Volhard identified crucial genes involved in the early patterning of the Drosophila embryo.
They systematically searched for embryonic lethal mutants, identifying 15 loci that altered the segmentation pattern of the embryo when mutated \citep{Nusslein-Volhard1980}, which they separated in tree groups based on their phenotype: "pattern duplication in each
segment (segment polarity mutants; six loci), pattern deletion in
alternating segments (pair-rule mutants; six loci) and deletion of
a group of adjacent segments (gap mutants; three loci) \citep{Nusslein-Volhard1980}".

All these genes form part of the A/P patterning cascade, whose hierarchical regulation is currently well known.

\subsubsection{Maternal effect genes}
The first A/P pattern of the embryo is determined in the egg chamber, during oogenesis.
The oocyte nucleus transports \textit{Gurken} protein close to the posterior part of the egg chamber. 
The follicle cells in that region receive the Gurken signal (Gurken is homologue of the vertebrate epidermal growth factor [EGF], see REF Neuman-Silberberg), which determines their fate as posterior cells.
This signal provokes the polarization of the microtubules in an A/P axis, that facilitates the transportation of mRNAs or proteins to specific parts of the oocyte.
Among these molecules are the mRNAs of the \textit{bicoid} and \textit{nanos}, which are transported to the anterior pole and posterior pole of the oocyte, respectively.
These and other genes, which are known as maternal effect genes, specify the A/P axis regulating specific target genes.

The maternal effect genes are classified in three different groups depending on their localization (anterior, posterior and terminal groups). Each group is briefly described below.

\paragraph{Anterior group}
After its anchorage to the anterior region of the embryo, the \textit{bicoid} mRNA is translated forming a gradient from the anterior to the posterior part of the embryo.
This protein determines the position of the anterior structures of the embryo acting as a \textit{morphogen}, i.e., different levels of Bicoid protein determines different cell fates in the anterior part of the embryo (REF Driever).
Bicoid is a transcription factor that regulates many target genes in a concentration-dependent manner. Foe example, expression of target genes in the head region require 1)high concentration of Bicoid and 2)the expression of \textit{hunchback}, a gene that is activated at moderate levels of Bicoid (REF Simpson Brose).  

\paragraph{Posterior group}
The \textit{nanos} mRNA, that is localized in the posterior region of the embryo, also generates a protein gradient. Nanos inhibits the translation of \textit{hunchback}(Tautz 1988) by forming a complex with other ubiquitous proteins in the embryo (REF Cho et al. 2006). The inhibition of \textit{hunchback} by Nanos causes an anterior to posterior gradient of the former. 
Another gene of the posterior group is the transcription factor \textit{caudal}. Contrary to \textit{nanos} or \textit{bicoid} mRNA, \textit{caudal} mRNA is distributed in the whole embryo.
Caudal gradient is formed by translation repression by the Bicoid protein. Caudal activates genes that determine the abdominal fate. 
The opossing gradients of Bicoid and Caudal will activate zygotic genes at different positions along the A/P axis of the blastoderm embryo.

\paragraph{Terminal group}
In mutants of terminal group genes the acron and telson are not formed (REF Klingler), which are the most anterior and posterior regions of the embryo, repectively. 
The boundaries of these structures are defined by the Torso signal. 
Torso is a tyrosine kinase receptor that, although is uniformly expressed along the surface membrane of early embryos, it is only activated at both poles (REF casanova).



\subsubsection{GAP gene network}

Gap genes were named like that as their mutants lacked some segments in the embryo, leaving a "gap" in the embryo. 
These genes constitute a dynamical gene network of transcription factors directly activated or repressed by the A/P gradients of the maternal effects genes.
Their expression consist of one or two broad domains in the embryo, whose boundaries are defined by five basic regulatory mechanisms ( REF Jaeger 2004):
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{enumerate}
\item Activation of gap genes by Bicoid and/or Caudal
\item Auto-activation
\item Strong repression between mutually exclusive gap genes
\item Repression between overlapping gap genes
\item Repression by Tolloid
\end{enumerate}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Some of the gap genes are \textit{hunchback}, \textit{knirps}, \textit{kr\"{u}ppel} and \textit{giant}.
Importantly, the patterning by gap gene network occurs in the late syncitial blastoderm stage, that corresponds to stage 4 and 5 in Campos-Ortega (REF).
This allows that the nuclei, still not surrounded by membranes, regulate each other expression with transcription factors.

\subsubsection{Pair-rule genes}

Gap genes control then the expression of the pair-rule genes, when the embryo is still in the syncitial blastoderm stage. 
Pair-rule genes were name like that as they are expressed in a regularly spaced striped pattern, each stripe corresponding to one parasegment.
Parasegments, which are considered the segmental unit in the embryo, do not correspond to the segments of the larva or adult, instead, parasegments and segments are out of phase (a segment is composed by anterior and posterior compartments, a parasegment is composed by the posterior compartment of a segment and the anterior compartment of the next segment;  REF Martinez Arias; lawrence).
%Mutants of a pair-rule genes lack the segments where the genes are normally expressed.

Pair-rule genes are divided in primary and secondary pair rule genes. The former are regulated by gap and maternal effect genes, while the latter are also regulated from the primary pair
rule genes (REF Wanninger).
The stripes of expression of pair-rule genes are modularly regulated by specific enhancers, each for one stripe or a pair of stripes.
This was firstly discovered in the stripe 2 of the \textit{even-skipped} (\textit{eve}) gene. 
Genetic studies determined that \textit{eve} stripe 2 is activated by Bicoid and Hunchback, while repressed by Giant and Kr\"{u}ppel (REF Small, Stanojevic).
Another pair rule gene that is expressed in an alternate manner to \textit{eve} is \textit{fushi-tarazu} (\textit{ftz}).


\subsubsection{Segment polarity genes}

The next step in the A/P patterning cascade is the activation of the segment polarity genes.
At this point blastoderm cells are already formed. Therefore, further pattern formation requires cell-cell communication (REF Gilbert book).
Segment polarity genes are regulated directly by gap and pair-rule genes and, as their name suggests, are expressed in the anterior or posterior side of the embryo para-segments.
The genes involved in this cell to cell communication are members of the Wnt/Wingless (Wg) and Hedgehog (Hh) signalling pathways.

The expression of Hh is regulated by the \textit{engrailed} gene, which in turn is activated is the cells where high levels of Even-skipped gene or Fushi Tarazu.
Additional regulatory inputs drive the expression of \textit{engrailed} in the anterior part of each parasegment.
Expression of \textit{wingless} is repressed by both Ftz and Eve and repressed by Odd-paired, so Wg is express only in one row of cells, adjacent to the cells expressing \textit{en}.
A forward feedback loop (REF Wannunger).
Interaction between the Wg and Hh signaling pathways, reinforce each other expression (REF Ingham; Heemskerk), maintaining the pattern formed by pair-rule genes and forming a stable boundary between anterior and posterior compartments of each para-segment.



  


\subsubsection{Hox genes}

\subsection{Fate map}


\subsection{Adaptation in the embryo}


\subsection{Hour-glass model in Drosophila}
