
%
%\subsubsection{Adaptation in the embryo}
%
%Usually, the study of adaptation
%- general considerations of adaptation in the embryo
%
%- example from marine animals
%- Whyte
%- entrenchment
%- adaptive changes that influence adult/larva characters
%- measurements of adaptation in the embryo in drosophila
%- heatshock, developmental time, Dn/Ds.. etc.
%\subsubsection{Hourglass model}

%%% von Baer vs Hourglass
\label{hourglass}

As I described briefly in section \ref{vonBaer}, von Baer stated in his "laws" that within a group of animals the general characteristics appear earlier in development, while the most special appear in late development \citep{vonBaer1828uber}.
%although he recognized that the early stages were different (Gilbert 2010)
This would lead to low morphological variation at early development, gradually increasing as development proceeds.

Other authors \citep{Medawar1954,Slack1993,Duboule1994,Raff1996} proposed an alternative pattern in which there is great variation in early and late development, while the mid-development would show less variation.
This pattern of variation (or conservation) has been called `phylotypic egg-timer' \citep{Duboule1994} and `developmental hourglass' \citep{Raff1996}.

% hourglass background
Duboule's concept of `phylotypic egg-timer' was based in the concept of `phylotypic stage' of \citet{Sander1983}, who coined this term to describe the convergence into a conserved segmented germ band stage in insects from very divergent early development \citep{Sander1996}.
In vertebrates, there has been controversy around what should be the phylotypic stage \citep{Ballard1981,Slack1993,Duboule1994}. \citet{Richardson1995} argued that indeed there is no single conserved stage in vertebrate's development and instead he proposed the term `phylotypic period' instead.

Initially, two explanations for the hourglass model were proposed.
Denis Duboule, after observing that the expression of the Hox genes seemed to coincide with the phylotypic stage, he considered that this could not be a coincidence and proposed that the activation of the Hox genes was the cause for the morphological invariance \citep{Duboule1994}.
In contrast, Rudolf A. Raff proposed that the phylotypic stage was the result of complex interaction between developmental modules at this stage  \citep{Raff1996}.

There is an ongoing discussion about whether the hourglass model (HG), the von Baer law
or some other pattern fits the divergence among developmental stages in phylogeny \citep{Richardson1997,Poe2004,Kalinka2012}.
	\nomenclature{HG}{Hourglass model}

Also, it is not clear if the HG, that seems to fit well in vertebrates and arthropods, would apply to other phyla \citep{Raff1996} \citep{Salazar-Ciudad2010}.
\citet{Salazar-Ciudad2010} has proposed that different patterns of variation throughout development in metazoan groups would correlate with different developmental types (a classification based on the relative use of signalling and morphogenetic events).

%% genetic hourglass
Recently, the HG have received support from different gene expression studies.
%Kalinka
\citet{Kalinka2010} used micro-arrays for six Drosophila species and quantified expression divergence at different developmental stages. They found that gene expression was most conserved during the extended germ-band stage (considered the phylotypic period) and that the non-synonymous divergence per site ($Dn$) correlated with their divergence measures.

They also showed that most genes fit best to models incorporating stabilizing selection and proposed that natural selection acts to conserve patterns of gene expression during mid-embryogenesis \citep{Kalinka2010}.

%Domazet & piasecka
The HG seems also to be reflected in the age of the transcriptome (mid-embryonic stage shows the older transcriptome; \citealp{Domazet-Loso2010}) and in the conservation of the regulatory regions (most conserved for genes expressed in mid-development; \citealp{Piasecka2013}).

%other based on DNA conservation
Studies measuring the conservation of genes at the DNA sequence level also seem to support the HG.
\citet{Davis2005} assessed whether proteins expressed at different times during \textit{D.  melanogaster} development varied systematically in their rates of evolution (comparing with \textit{D. pseudoobscura}) and found that proteins expressed early in development and particularly during mid-late embryonic development evolve slower.  
This suggests, according to the authors, that embryonic stages from 12 to 22 hours are
highly conserved between \textit{D. melanogaster} and \textit{D. pseudoobscura}, which is consistent with the HG.
%
In a similar study, \citet{Mensch2013} calculated the dN/dS ratio for more than 2,000 genes among six \textit{Drosophila} species, separating genes in three categories: maternal genes (genes whose products are left by the mother in the egg), genes expressed in early development and genes expressed in late development. They found that maternal genes and lately expressed zygotic genes show higher dN/dS ratios (i.e., are less conserved) than early expressed zygotic genes.
Finally, it has also been found that genes expressed in the adult have higher dN/dS ratios than genes expressed in the pupa and those of the pupa have higher dN/dS ratios that those expressed in the embryo \citep{Artieri2009}.

Some limitations of these last studies is that they classify the genes in a few broad temporal categories that do not permit to precisely determine the temporal dynamics of conservation and that are based only in divergence data (dN/dS ratios between two species).
A study that integrates polymorphism data from natural populations would improve the evolutionary interpretation of these patterns, as it would allow to estimate what proportion of the dN are adaptive (as explained in section \ref{alpha}).
Measuring adaptation is specially relevant as some authors have argued that the HG is caused by different selection pressures in early and late development \citep{Slack1993,Kalinka2012,Wray2000}

%%inter phyla comparison

%Recently, a study reported an inverted hourglass pattern
%(i.e., early and late conservation and mid-development divergence) when comparing developmental transcriptomes of ten different species, each from a different phyla [21](REF).