
To estimate adaptation during \textit{D. melanogaster} embryogenesis, the DFE-alpha method and software were used (see section \ref{alpha}; \citealp{Eyre-Walker2009}), which infer adaptation combining polymorphism and divergence data.

The DFE-alpha software (DFE-alpha, \citealp{Eyre-Walker2009}; see below) requires that all sites to have been sampled in the same number of chromosomes. Therefore, the original DGRP dataset was reduced to from 168 to 128 isogenic lines by randomly sampling the polymorphisms at each site without replacement. Residual heterozygous sites and sites with no quality value were excluded from the analysis.
%
This software estimate several parameters (e.g., $\alpha$ and $\omega_{\alpha}$) from a set of genes as estimates based on single genes can be affected by the lack of segregating (divergent) sites. Therefore, in each analysis a group of genes was randomly sampled (bootstrap with replacement) (see studies III and IV). 
As neutral reference the positions 8-30 of short introns ($\leq$ 65 bp) were used (as in \citealp{Heyn2014}). For validation, 4-fold degenerate sites were also used.

The release 5 of the Berkeley Drosophila Genome Project was used as the reference genome (http://www.fruitfly.org/sequence/release5genomic.shtml/). The divergence statistics were estimated from a multiple genomic alignment between DGRP lines and \textit{D. yakuba} BDGP 5 coordinates (from http://popdrowser.uab.cat; \citealp{Ramia2012}).
The number of sites and substitutions and the folded site frequency spectrum (SFS) were computed using an ad hoc Perl script.

Ortholog genes between \textit{D. yakuba} and \textit{D. melanogaster} were obtained from FlyBase (http://flybase.org/). \textit{D. yakuba} was used as outgroup species as, due to the time since their divergence, there is less chance of ancestral polymorphism contributing to divergence, diminishing the effect of low divergence affecting the estimates of adaptive evolution \citep{Keightley2012}.